nemo/core/neural_types/neural_type.py

```diff
@@ -223,6 +223,7 @@ def __init__(self, producer, producer_args, output_port_name, ntype=None):
         self._step_number = AppState().active_graph.step_number
```

Comment:
I didn't see the need to store which graph is the "owner" of a given tensor.

But how about adding this information to NmTensor? Seems very easy:
self._graph = AppState().active_graph.name
(+ property def graph(self))

So then accessing all tensor in a given "graph_a":
AppState().graphs["graph_a"].tensors_list

Moreover, this will also give you access to "graph execution engine", which, at its core, is graph.steps -> note that e.g. DL should be step 0 (in a "valid executable graph"), so you won't even need to copy the "inverse graph traverse" ;)

So injection of this single line seems to be quite powerful ;)

Reply:
Ok, clearly you don't need it. I will add it when I will be working on other graph actions. Closing.